x-pack/solutions/observability/plugins/profiling/public/views/storage_explorer/index.tsx

```diff
@@ -63,6 +64,17 @@ export function StorageExplorerView() {
     storageExplorerSummaryState.data?.totalNumberOfDistinctProbabilisticValues || 0;
   const hasDistinctProbabilisticValues = totalNumberOfDistinctProbabilisticValues > 1;
 
+  const { onPageReady } = usePerformanceContext();
+
+  if (storageExplorerSummaryState.status === AsyncStatus.Settled) {
+    onPageReady({
+      meta: {
+        rangeFrom,
+        rangeTo,
+      },
+    });
```

Comment:
I think the page will be ready when  `storageExplorerSummaryState` is complete and the default open tab content also finishes loading. In this case, `host_breakdown`. Wdyt?

Reply:
Good catch, I agree. I'll add a status check for the host breakdown.

Reply:
I've moved the onPageReady call into the host_breakdown component. I'm not super jazzed about it, but pulling the host_breakdown fetch into the storage_explorer looked like a mess due to the tab state changes. One issue with this solution is the onPageReady will be called every time the tab is toggled, I tried wrapping the call in a useMemo, but it looks like the api-call is also refreshed, so it seems like more of an issue with the component itself. Is this due to the fact the time window is technically changed from the last fetch? Is this ok to leave alone? 
Do you think this solution is alright, or do you have an alternative suggestion?
If this looks ok, then it might be worthwhile to add telemetry to the data breakdown tab as well.

Reply:
Hey Bryce, thanks for the changes. I think it's fine to call `onPageReady` more than once. It will only register one event per page view: https://github.com/crespocarlos/kibana/blob/main/src/platform/packages/shared/kbn-ebt-tools/src/performance_metrics/context/use_page_ready.ts

You could also test this by following this [comment](https://github.com/elastic/kibana/issues/205396#issuecomment-2618567529) and checking on Discover whether the event is registered on every tab switch.

Should new events be registered when switching tabs, we'll need to see what we can do.